version-check.js

```diff
@@ -0,0 +1,119 @@
+const https = require('https');
+const exec = require('child_process').exec;
+
+// TODO: switch to develop branch
+const mappingsSrc = 'https://raw.githubusercontent.com/service-mocker/service-mocker/ts-compatibility-check/ts-version-map.json';
+
+const MOCKER_MATCHED = 1 << 0;
+const TS_MATCHED = 1 << 1;
+
+function installDeps() {
+  const deps = Array.prototype.slice.call(arguments);
+  const missed = [];
+
+  deps.forEach((pkg) => {
+    try {
+      require.resolve(pkg);
+    } catch (e) {
+      missed.push(pkg);
+    }
+  });
+
+  if (!missed.length) {
+    return Promise.resolve();
+  }
+
+  return new Promise((resolve, reject) => {
+    exec(`npm install ${missed.join(' ')}`, (err, stdout, stderr) => {
+      if (err || stderr) {
+        reject(err || stderr);
+      } else {
+        resolve(stdout);
+      }
+    });
+  });
+}
+
+function fetchRemote() {
+  return new Promise((resolve, reject) => {
+    https.get(mappingsSrc, (res) => {
+      let rawData = '';
+
+      res.on('data', (chunk) => {
+        rawData += chunk;
+      });
+
+      res.on('end', () => {
+        try {
+          resolve(JSON.parse(rawData));
+        } catch (e) {
+          reject(e);
+        }
+      });
+    }).on('error', reject);
+  });
+}
+
+function compareVer(versions) {
+  const mockerVersion = require('./package.json').version;
+  const tsVersion = require('typescript').version;
+
+  return fetchRemote()
+    .then((data) => {
+      const semver = require('semver');
+
+      console.log('Compatibility List:', JSON.stringify(data.mappings, null, 2));
+
+      let finalState = 0;
+      let error = null;
+
+      data.mappings.some((mapping) => {
+        let state = 0;
+
+        if (semver.satisfies(mockerVersion, mapping['service-mocker'])) {
+          state |= MOCKER_MATCHED;
+          finalState |= MOCKER_MATCHED;
+        }
+
+        if (semver.satisfies(tsVersion, mapping['typescript'])) {
+          state |= TS_MATCHED;
+          finalState |= TS_MATCHED;
+        }
+
+        switch (state) {
+          case MOCKER_MATCHED & TS_MATCHED:
```

Comment:
`case MOCKER_MATCHED & TS_MATCHED` means nothing matched?

I think it should be `MOCKER_MATCHED | TS_MATCHED` to match both.